pkg/appgw/configbuilder.go

```diff
@@ -182,19 +183,37 @@ func generateBackendID(ingress *v1beta1.Ingress, rule *v1beta1.IngressRule, path
 	}
 }
 
-func generateListenerID(rule *v1beta1.IngressRule, protocol n.ApplicationGatewayProtocol, overridePort *Port, usePrivateIP bool) listenerIdentifier {
+func generateListenerID(ingress *v1beta1.Ingress, rule *v1beta1.IngressRule, protocol n.ApplicationGatewayProtocol, overridePort *Port, usePrivateIP bool) listenerIdentifier {
 	frontendPort := Port(80)
 	if protocol == n.HTTPS {
 		frontendPort = Port(443)
 	}
 	if overridePort != nil {
 		frontendPort = *overridePort
 	}
+
 	listenerID := listenerIdentifier{
 		FrontendPort: frontendPort,
-		HostName:     rule.Host,
 		UsePrivateIP: usePrivateIP,
 	}
+
+	// Populate the hostnames property
+	var hostnames []string
+	if rule.Host != "" {
+		hostnames = append(hostnames, rule.Host)
+	}
+
+	if extendedHostNames, err := annotations.GetHostNameExtensions(ingress, rule.Host); err == nil {
+		if extendedHostNames != nil {
```

Comment:
run `go fmt ./...` to correct the spacing

Reply:
Done